Classes/Service/ImageResizer.php

```diff
@@ -271,6 +271,21 @@ public function processFile($fileName, $targetFileName = '', $targetDirectory =
             }
         }
 
+        if (
+            isset($ruleset['max_size'])
+            && $ruleset['max_size'] > 0
+            && isset($metadata['width'])
+            && $metadata['width'] > 0
+            && isset($metadata['height'])
+            && $metadata['height'] > 0
+            && $metadata['width'] * $metadata['height'] > $ruleset['max_size']
```

Comment:
could you please elaborate on this `width * height > max_size` ?

Reply:
This checks if the actual image size (number of pixels, height times width) is bigger than the configured maximal image size. This is in addition to the already existing limits for height / width / file size.